### PR TITLE
fieldManager: Ignore and log all errors when updating managedFields

### LINKED ISSUE #90904: fieldManager errors can fail create/update/non-apply-patch API calls
**What happened**:

FieldManager errors were encountered in an Update API request (xref https://github.com/openfaas/faas-netes/issues/630#issuecomment-625440515):
```
Unexpected status: 500, message: failed to update object (Update for openfaas.com/v1, Kind=Function) managed fields: failed to convert live object to proper version: /, Kind= is unstructured and is not suitable for converting to "openfaas.com/v1"
```

This error blocked an update request that worked in prior Kubernetes versions.

**What you expected to happen**:

The particular error that occurred there should be tracked down and resolved separately, but I would expect field manager errors only block apply requests, not create/update/non-apply-patch requests.

In these non-apply handler locations, fieldManager errors are a "should not happen" type of failure. If a fieldManager-related error is encountered, I'd expect to log the error (and perhaps bump a server-side-apply-error metric), but to then recover and allow the create/update/patch request to continue (potentially by resetting the managedFields to avoid persisting bad data):

```
grep "Manager.Update(" staging/src/k8s.io/apiserver/pkg/endpoints/handlers/*.go
staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go:				obj, err = scope.FieldManager.Update(liveObj, obj, managerOrUserAgent(options.FieldManager, req.UserAgent()))
staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go:		if objToUpdate, err = p.fieldManager.Update(currentObject, objToUpdate, managerOrUserAgent(p.options.FieldManager, p.userAgent)); err != nil {
staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go:		if newObj, err = p.fieldManager.Update(currentObject, newObj, managerOrUserAgent(p.options.FieldManager, p.userAgent)); err != nil {
staging/src/k8s.io/apiserver/pkg/endpoints/handlers/update.go:					obj, err := scope.FieldManager.Update(liveObj, newObj, managerOrUserAgent(options.FieldManager, req.UserAgent()))
```

/priority critical-urgent
/sig api-machinery
/assign @apelisse

### PR DESCRIPTION
**Pull request by apelisse**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:
Fixes #90904

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go [hunk 1]
```diff
@@ -138,6 +144,20 @@ func (f *FieldManager) Update(liveObj, newObj runtime.Object, manager string) (o
...
+// UpdateNoErrors is the same as Update, but it will not return
+// errors. If an error happens, the object is returned unchanged as if
+// this had never happened.
+func (f *FieldManager) UpdateNoErrors(liveObj, newObj runtime.Object, manager string) runtime.Object {
+	obj, err := f.Update(liveObj, newObj, manager)
+	if err != nil {
+		atMostEverySecond.Do(func() {
+			klog.Errorf("[SHOULD NOT HAPPEN] failed to update managedFields: %v", err)
+		})
+		return newObj
```
Comment (apelisse) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go:156]: Maybe add `internal.RemoveObjectManagedFields(newObj)` before?

Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go:156]: explicitly clear managedFields on newObj before returning here and document the behavior

File: staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go [hunk 2]
```diff
@@ -138,6 +144,20 @@ func (f *FieldManager) Update(liveObj, newObj runtime.Object, manager string) (o
...
 }
 
+// UpdateNoErrors is the same as Update, but it will not return
+// errors. If an error happens, the object is returned unchanged as if
+// this had never happened.
+func (f *FieldManager) UpdateNoErrors(liveObj, newObj runtime.Object, manager string) runtime.Object {
+	obj, err := f.Update(liveObj, newObj, manager)
+	if err != nil {
+		atMostEverySecond.Do(func() {
+			klog.Errorf("[SHOULD NOT HAPPEN] failed to update managedFields: %v", err)
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/endpoints/handlers/fieldmanager/fieldmanager.go:154]: include the type of newObj in the message?

File: test/integration/apiserver/apply/apply_test.go [hunk 3]
```diff
@@ -1212,7 +1212,7 @@ func TestClearManagedFieldsWithMergePatch(t *testing.T) {
 		Namespace("default").
 		Resource("configmaps").
 		Name("test-cm").
-		Body([]byte(`{"metadata":{"managedFields": [{}]}}`)).Do(context.TODO()).Get()
```
Comment (liggitt) [inline test/integration/apiserver/apply/apply_test.go:1215]: do we have tests exercising clearing managed fields using all of the following scenarios:
* resource types: built-in and custom
* patch types: merge, json, and strategic-merge
* managedFields content: `[{}]` and `[]`

**Comment by liggitt** [comment]
were you anticipating bringing in https://github.com/kubernetes-sigs/structured-merge-diff/pull/164 in this PR as well, to get a single k/k PR for backporting?

**Comment by liggitt** [comment]
replaced by https://github.com/kubernetes/kubernetes/pull/91690

**Pull request closed without merging**